src/main/java/net/coderbot/iris/shaderpack/ShaderProperties.java

```diff
@@ -32,9 +33,40 @@ public ShaderProperties(Properties properties) {
 				viewportScaleOverrides.put(pass, scale);
 			}
 		});
+		this.properties = ImmutableProperties.of(properties);
 	}
 
 	public static ShaderProperties empty() {
 		return new ShaderProperties();
 	}
+
+	public Properties asProperties() {
+		return properties;
+	}
+
+	private static class ImmutableProperties extends Properties {
+		@Override
+		public synchronized Object setProperty(String key, String value) {
```

Comment:
I don't see why you need to synchronize methods that are `_ -> fail`

Reply:
I never noticed that it was implemented as a synchronized method... good catch, will be fixed